mewarpx/tests/test_field_diagnostic.py

```diff
@@ -92,32 +92,51 @@ def test_field_diag(plot_on_diag_steps):
         init_inert_gas=True,
         init_neutral_plasma=True,
         init_field_diag=True,
-        init_simcontrol=True,
         init_warpx=True,
         init_simulation=True
     )
-    while run.control.check_criteria():
-        mwxrun.simulation.step()
+    mwxrun.simulation.step()
 
     # verify that the plot images were created.
     if plot_on_diag_steps:
-        print("Verifying that all plots were created...", flush=True)
-        assert len(glob.glob(f"diags/fields/Electrostatic_potential_*.npy")) == 5
-        assert len(glob.glob(f"diags/fields/Electrostatic_potential_*.png")) == 5
+        print("Verifying that all data and plot files were created...")
```

Comment:
We should put flush=True for all these diagnostic print statements, or use the logger to immediately print them. Otherwise, the statements will not print at the expected times and will not indicate where the failure is if the test crashes.

Reply:
Well these ones will cause the test to terminate if any of those assertions are true, so the final print statement will still show where the test failed.